### DISCUSSION TITLE
Is there any way to reroute editors connections?

### DISCUSSION THREAD

**Discussion by dabulina** [cables-gl/cables_docs#211] [Feature Requests] [answered]
By defoult cables creates shortest visual connection, but often it creates mess in editor. Maybe there is a way that i don't know how to reroute connections.

**Comment by pandrr** [comment, marked as answer]
you can use number op or triggerExtender, but no not really (yet)

**Reply by dabulina** [reply]
Yea, triggerExtender is limiter and consumes a lot of space.
Today again I'm dreaming about double click reRoute option like Unreal have.

**Comment by franckaubry** [comment]
I've been dreaming of a "Null op" for this.
Another way to solve these kinds of workflows is by adding the possibilities to insert extra points in the cables. Personally I often find that this adds a lot of extra housekeeping when (re)organizing ops, but other people swear to this. (The screenshot is an example from the Houdini interface)

![image](https://user-images.githubusercontent.com/97125854/161508658-70832a83-6d12-4030-8df0-932a03b7bd8a.png)

**Comment by dabulina** [comment]
look: this simple math goes into complex geo 👯‍♀️ 
is it possible to customize the cables UI with some magic patch?
<img width="2048" alt="Screenshot 2022-07-17 at 18 01 54" src="https://user-images.githubusercontent.com/4573092/179410192-3a07f70a-8e9c-4eca-a4c0-4a9e78c9f73e.png">
#

**Reply by steam0r** [reply]
you could put this into a subpatch, right? or write a userop that does the math (but then you have the same "mess" in code. to clean up these connections visually you should maybe work with variables and, for example, store the canvasinfo things in variables...that would clean up these long cables that cross everything.

https://cables.gl/op/Ops.Vars.VarSetNumber_v2

**Reply by dabulina** [reply]
Yea, Variables, SubPatch, CustomOP are solutions. When I'm working in teams with students, it's super handy to leave an unfolded structure that others can easily follow <3 I hope to be heard from developers. Maybe one day, we will have "reroute" magic. Cables are a brilliant piece of software!

**Reply by steam0r** [reply]
still not sure what this reroute thing would do that we do not have yet? :) bend cables? can you maybe illustrate that somehow?

also, above, you are saying triggerextender is "a limiter", what do you mean by that? are you mixing this up with triggerlimiter or delayedtrigger?
